interface/src/App.js

```diff
@@ -1,149 +1,214 @@
-    clearInterval(intervalRef.current);
-    setIsRecording(false)
+  function startRecording() {
+    setIsStreamPending(true);
+    navigator.mediaDevices
+      .getUserMedia({ audio: true })
+      .then(function (s) {
+        streamRef.current = s;
+
+        // Create a new WebSocket connection.
+        socketRef.current = new io.connect("http://0.0.0.0:8000", {
+          transports: ["websocket"],
+        });
+        const config = {
+          language: selectedLanguage,
+          model: selectedModel,
+        };
+        socketRef.current.emit("startWhispering", config);
+
+        // When the WebSocket connection is open, start sending the audio data.
+        socketRef.current.on("whisperingStarted", function () {
+          setIsStreamPending(false);
+          setIsRecording(true);
+          audioContextRef.current = new (window.AudioContext ||
+            window.webkitAudioContext)({
+            sampleRate: MIC_SAMPLE_RATE,
+          });
+          var source = audioContextRef.current.createMediaStreamSource(
+            streamRef.current
+          );
+          var processor = audioContextRef.current.createScriptProcessor(
+            BLOCK_SIZE,
+            1,
+            1
+          );
+          source.connect(processor);
+          processor.connect(audioContextRef.current.destination);
+
+          processor.onaudioprocess = function (event) {
+            var data = event.inputBuffer.getChannelData(0);
+            setAudioData(new Float32Array(data));
+
+            socketRef.current.emit("audioChunk", b64encode(data));
+          };
+        });
+
+        socketRef.current.on(
+          "transcriptionDataAvailable",
+          (transcriptionData) => {
+            console.log(`transcriptionData: ${transcriptionData}`);
+            handleTranscribedData(transcriptionData);
+          }
+        );
+      })
+      .catch(function (error) {
+        console.error("Error getting microphone input:", error);
+      });
   }
 
-  function transcribeRecording(recordedBlob) {
-    const headers = {
-      "content-type": "multipart/form-data",
-    };
-    const formData = new FormData();
-    formData.append("language", selectedLangRef.current)
-    formData.append("model_size", modelOptions[selectedModelRef.current])
-    formData.append("audio_data", recordedBlob.blob, 'temp_recording');
-    axios.post("http://0.0.0.0:8000/transcribe", formData, { headers })
-      .then((res) => {
-        setTranscribedData(oldData => [...oldData, res.data])
-        setIsTranscribing(false)
-        intervalRef.current = setInterval(transcribeInterim, transcribeTimeout * 1000)
-      });
-      
-      if (!stopTranscriptionSessionRef.current){
-        setIsRecording(true)    
-      }
+  function stopRecording() {
+    setIsStreamPending(true);
+    socketRef.current.emit("stopWhispering");
+    streamRef.current.getTracks().forEach((track) => track.stop());
+    audioContextRef.current.close();
+    setAudioData([]);
```

Comment:
What about the data that was sent until this moment and is still being processed by the backend? Will it be appended even after the user clicked Stop?

Reply:
P.S. it should get appended if I wasn't clear :)